plans/action/configure.pp

```diff
@@ -74,10 +74,16 @@
   }
 
   if $arch['high-availability'] {
+    $topology = $compiler_targets.empty ? {
+      true    => 'mono',
+      default => 'mono-with-compile',
+    }
+
```

Comment:
I would like to see a native function that wraps all this.

`$configured_topology = topology($arch, $compiler_targets)`

or 

```
run_task('peadm::provision_replica', $master_target,
       master_replica => $master_replica_target.peadm::target_name(),
       token_file     => $token_file,
       topology       =>  topology($arch, $compiler_targets),
```

Reply:
What about expanding peadm::validate_architecture instead?
Currently it only returns 'high-availability' and  'architecture' , we can add 'topology' there?

Reply:
What is the difference between topology vs architecture?

Reply:
`architecture` is peadm's nomenclature - standard, large and extra-large
`topology` is PE replica provisioning parameter - mono or 'mono-with-compile'

Currently, peadm configures replica for 'standard' and 'high-availability' only as a DR target, but it doesn't provide high-availability configuration, since agents have only one server listed in server_list